### PR TITLE
allow exporting a number of recent chain state trees

### PR DESCRIPTION
**Pull request by whyrusleeping**
Gets us closer to being able to 'sync' faster.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
`lotus chain export --recent-stateroots=600 snapshot.car`

wait like, 20 minutes, observe creation of 10gb file.

`lotus daemon --snapshot=snapshot.car`

Skip entire sync process!

File: cmd/lotus/daemon.go [hunk 1]
```diff
@@ -191,13 +195,23 @@ var DaemonCmd = &cli.Command{
 		}
 
 		chainfile := cctx.String("import-chain")
-		if chainfile != "" {
+		snapshot := cctx.String("checkpoint")
+		if chainfile != "" || snapshot != "" {
+			if chainfile != "" && snapshot != "" {
+				return fmt.Errorf("cannot specify both 'snapshot' and 'import-chain'")
```
Comment (Kubuxu) [inline cmd/lotus/daemon.go:201]:
```suggestion
				return fmt.Errorf("cannot specify both 'checkpoint' and 'import-chain'")
```

Comment (jbenet) [reply cmd/lotus/daemon.go:201]: - `checkpoint` is a term that carries social consensus connotations. 
- i think snapshot is the right term for the exported state. (ie not all snapshots one may load should be thought of as checkpoints)
- maybe the daemon load flag should be `lotus daemon --load-state=snapshot.car` or `lotus daemon --load-snapshot=snapshot.car`

**Comment by jbenet** [comment]
Thanks so much -- this will help so many people trying to sync the chain, specially from scratch 🙏

File: cli/chain.go [hunk 2]
```diff
@@ -872,6 +876,8 @@ var chainExportCmd = &cli.Command{
 			return fmt.Errorf("must specify filename to export chain to")
 		}
 
+		rsrs := cctx.Int64("recent-stateroots")
```
Comment (Kubuxu) [inline cli/chain.go:879]: We should probably require `recent-stateroots` to be greater than finality because some lookbacks are that long.

**Review by magik6k — APPROVED** [review]

**Pull request merged**